src/mono/mono/metadata/bundled-resources.c

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#include <glib.h>
+#include <stdbool.h>
+
+#include "bundled-resources-internals.h"
```

Comment:
Ah, I wasn't sure if the `<mono/metadata/...>` was more for the public apis like those under `src/native/public/mono/metadata/assembly.h`. I saw in `src/mono/mono/metadata/assembly.c`, some internals seemed to be wrapped with `""`, and just followed some of those examples. Should both the internals and the public apis both be `<mono/metadata/...>`?

Reply:
We have different ways of including in runtime evolved over time. Runtime build will always setup include folders so we can include based out of mono directory, there are also include folders setup to eglib, sgen, public include folder plus and folders where we can generate build output that will be included by source files. Using <> around include will make sure compiler will use the search paths to locate the include files, while using "" will start to search using the current compiled source current directory and if not found there, fallback to include paths. Since we include based on a path in these scenarios, compiler will never find the file relative to the compiled source file, so in those case we should use <>. In this case you can argue that you include a file that sits next to the source file, so include it using "" and just the file name will then look locally first and if not found, use include paths, so will be a more effective way to include a file from the same directory as the source file.

So if a include files is guaranteed to sit next to the source file you can always do "myfile.h" and that will be the most effective way to include the local header file. If you include it using a relative path (when header sits in a different location in the source tree), you probably need to use <> to leverage the include paths. If the header is or might end up as a public header, include it using <> to leverage the include paths. So if you know your source will be in for example metadata folder, you can include all headers that also sits in metadata folder using "", that is done to some degree in runtime as well, but we have more and more includes that leverage relative paths, even for headers from the same "package", since that make things a little more flexible.